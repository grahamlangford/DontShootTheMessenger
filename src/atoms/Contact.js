import React from 'react'
import PropTypes from 'prop-types'
import { Dimensions, StyleSheet, Text, View } from 'react-native'

const Contact = ({ onPress, item }) => {
  return (
    <Text style={styles.listItem} onPress={() => onPress(item)}>
      {item}
    </Text>
  )
}

const styles = StyleSheet.create({
  listItem: {
    height: 48,
    width: Dimensions.get('window').width,
    textAlignVertical: 'center',
    paddingLeft: 16
  }
})

Contact.propTypes = {
  onPress: PropTypes.func.isRequired,
  item: PropTypes.string.isRequired
}

export default Contact
