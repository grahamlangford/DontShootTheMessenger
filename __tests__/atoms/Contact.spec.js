// __tests__/atoms/Contact.js
import { Text } from 'react-native'
import Contact from '../../src/atoms/Contact'
import { React, shallow } from '../helpers/specHelper'

describe('Contact.js', () => {
  let wrapper
  const onPressMock = jest.fn()
  const item = 'Picard'
  beforeEach(() => {
    jest.resetModules
    wrapper = shallow(<Contact onPress={onPressMock} item={item} />)
  })

  it('renders', () => {
    expect(wrapper).toBeTruthy()
  })

  it('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
