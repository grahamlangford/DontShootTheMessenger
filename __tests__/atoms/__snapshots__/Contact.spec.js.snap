// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact.js matches the snapshot 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  onPress={[Function]}
  style={
    Object {
      "height": 48,
      "paddingLeft": 16,
      "textAlignVertical": "center",
      "width": 750,
    }
  }
>
  Picard
</Text>
`;
